src/PerformanceMonitor/PerformanceMonitor.BlockProperties.cs

```diff
@@ -0,0 +1,242 @@
+        private void OnBlockTimestampCommand(uint blockCount)
+        {
+            uint desiredCount = blockCount;
+
+            if (desiredCount < 1)
+            {
+                Console.WriteLine("Minimum 1 block");
+                return;
+            }
+
+            if (desiredCount > 1000)
+            {
+                Console.WriteLine("Maximum 1000 blocks");
+                return;
+            }
+
+            using (var snapshot = Blockchain.Singleton.GetSnapshot())
+            {
+                var block = GetBlock(snapshot.CurrentBlockHash.ToString());
+                var blocksTime = snapshot.GetBlocksTimestamp(desiredCount, block);
+
+                if (blocksTime.Count == 0)
+                {
+                    Console.WriteLine("No blocks were found");
+                }
+                else
+                {
+                    Console.WriteLine($"{"Index",10}\tTimestamp");
+                    foreach (var blockTime in blocksTime)
+                    {
+                        var timestamp = DateTime.UnixEpoch.AddMilliseconds(blockTime.Value);
+                        timestamp = TimeZoneInfo.ConvertTimeFromUtc(timestamp, TimeZoneInfo.Local);
+                        Console.WriteLine($"{blockTime.Key,10}\t{timestamp.ToShortDateString()} {timestamp.ToLongTimeString()}");
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the block time in seconds for each of the latest blocks.
+        /// </summary>
+        /// <returns>
+        /// Returns the average time per block in milliseconds
+        /// </returns>
+        [RpcMethod]
+        public JObject GetBlocksTime(JArray _params)
+        {
+            if (_params.Count != 2)
+            {
+                throw new RpcException(-32602, "Invalid params");
+            }
+
+            if (!uint.TryParse(_params[0].AsString(), out uint desiredCount))
+            {
+                throw new RpcException(-32602, "Invalid params");
+            }
+
+            if (desiredCount < 1)
+            {
+                throw new RpcException(-100, "Minimum 1 block");
+            }
+
+            // It is currently limited to query blocks generated in the last 24 hours
+            uint maxNBlocksPerDay = 24 * 60 * 60 / (Blockchain.MillisecondsPerBlock / 1000);
+            if (desiredCount > maxNBlocksPerDay)
+            {
+                throw new RpcException(-100, maxNBlocksPerDay.ToString("Maximum 0 blocks"));
+            }
+
+            if (!uint.TryParse(_params[1].AsString(), out uint height) || desiredCount > height)
+            {
+                throw new RpcException(-32602, "Invalid params");
+            }
+
+            using (var snapshot = Blockchain.Singleton.GetSnapshot())
+            {
+                var block = GetBlock(height.ToString());
+                var blocksTime = snapshot.GetBlocksTimestamp(desiredCount, block);
```

Comment:
Did you test the time it spend with the maximum values?

Reply:
I've tested the cli's method that returns the same information. The cli's method is a bit lazier than the rpc's because of the printed messages. In my PC I got around 0.5 seconds for getting and printing the information.
![image](https://user-images.githubusercontent.com/19419485/87087436-744e3c80-c209-11ea-9abb-0e36b4aa5898.png)
![image](https://user-images.githubusercontent.com/19419485/87087437-757f6980-c209-11ea-9c78-44f43a980775.png)

Reply:
I think that it should be lower than 1000 in rpc, in console it's ok

Reply:
Sure, I have no problem decreasing it.